File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/Room.java
package dev.skyherobrine.service.models.mongodb;

import com.fasterxml.jackson.annotation.JsonFormat;
import dev.skyherobrine.service.models.mariadb.Appointment;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "rooms")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class Room {
    @MongoId
    private Long id;
    private String roomId;
    private Appointment appointment;
    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(targetType = FieldType.DATE_TIME)
    private LocalDateTime createdAt;
}

File: healthcare_app_server/chat_service/models/room.js
class Room {
    constructor(id, appointment, roomId, createdAt) {
        this._id = id;
        this._appointment = appointment;
        this._roomId = roomId;
        this._createdAt = createdAt;
    }


    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get appointment() {
        return this._appointment;
    }

    set appointment(value) {
        this._appointment = value;
    }

    get roomId() {
        return this._roomId;
    }

    set roomId(value) {
        this._roomId = value;
    }

    get createdAt() {
        return this._createdAt;
    }

    set createdAt(value) {
        this._createdAt = value;
    }
}

module.exports = Room;